src/main/java/core/basesyntax/BinaryTree.java

```diff
@@ -22,26 +22,61 @@ public void setRoot(Node root) {
     /* Given a binary tree, print its nodes according to the 
       "bottom-up" post order traversal. */
     public List<Node> getNodesPostOrder(Node node) {
-        System.out.println("Please provide your implementation");
-
-        return new ArrayList<>();
+        if (node == null) {
+            return new ArrayList<>();
+        }
+        List<Node> list = new ArrayList<>();
+        list.addAll(getNodesPostOrder(node.getLeft()));
+        list.addAll(getNodesPostOrder(node.getRight()));
+        list.add(node);
+        return list;
     }
 
     /* Given a binary tree, print its nodes in inorder*/
     public List<Node> getNodesInorder(Node node) {
-        System.out.println("Please provide your implementation");
-        return new ArrayList<>();
+        if (node == null) {
+            return new ArrayList<>();
+        }
+        List<Node> list = new ArrayList<>();
+        list.addAll(getNodesInorder(node.getLeft()));
+        list.add(node);
+        list.addAll(getNodesInorder(node.getRight()));
+        return list;
     }
 
     /* Given a binary tree, print its nodes in pre order*/
     public List<Node> getNodesPreOrder(Node node) {
-        System.out.println("Please provide your implementation");
-        return new ArrayList<>();
+        if (node == null) {
+            return new ArrayList<>();
+        }
+        List<Node> list = new ArrayList<>();
+        list.add(node);
+        list.addAll(getNodesPreOrder(node.getLeft()));
+        list.addAll(getNodesPreOrder(node.getRight()));
+        return list;
     }
 
     /* Given a binary tree, print its nodes in pre order*/
     public List<Node> getNodesBreadthFirst(Node node) {
```

Comment:
check if your algorithm works on 4+ level tree

try to use Queue for this method